### PR TITLE
feat!: re-create component for Vaadin 22+

### PR DESCRIPTION
**Pull request by web-padawan**
##  ⚠️ Breaking changes

There are many breaking changes related to Vaadin 22 upgrade. Please make sure you bump `3.0.0` after merge!

1. Updated Vaadin dependencies to `^22.0.0` and changed to use new packages e.g. `@vaadin/component-base`,
2. Re-implemented to align with the latest Vaadin components, using slotted `<input>` / `<label>` / error message,
3. Removed `readonly-container` part. From now on, the actual `<input>` is used when `readonly` is set to true,
4. Renamed `token` component to `chip` as [Material design](https://material.io/components/chips) suggests (so we can extract this component later),
5. Significant changes to core styles, Lumo and Material themes intended to reduce custom styling overrides.

## 💅  Internal changes

1. Rewritten unit tests completely to test API and not internal methods. Also, changed to use `@web/test-runner`,
2. Updated `eslint` and `stylelint` config to use the same configuration as Vaadin components, including Prettier,
3. Improved TypeScript definitions, added `tsconfig.json` and `lint:types` script to check that typings are correct,
4. Updated demos: run Prettier, removed polyfill, changed `<script>` tags to not use `WebComponentsReady` event,
5. Updated Git hooks config to use `simple-git-hooks` (lightweight alternative to `husky`) and `lint-staged`,
6. Added `"module"` entrypoint to `package.json` to indicate that package uses ES modules for some bundlers,
7. Replaced Travis build config with corresponding GitHub actions workflow, to run linters and tests for each PR,
8. Updated `.gitignore` and committed `package-lock.json` to the repo for faster and consistent builds in CI.

### CODE REVIEW

**Comment by srijken** [comment]
@web-padawan I had to edit some devDependencies to get your branch to work, otherwise `npm install` fails.

```diff
--- a/package.json
+++ b/package.json
@@ -61,13 +61,13 @@
   "devDependencies": {
     "@polymer/iron-demo-helpers": "^3.0.0",
     "@webcomponents/webcomponentsjs": "^2.0.0",
-    "eslint": "^6.0.1",
-    "eslint-config-vaadin": "^0.2.7",
+    "eslint": "^7.0.0",
+    "eslint-config-vaadin": "^0.4.0",
     "eslint-plugin-html": "^6.0.0",
     "husky": "^3.0.8",
     "npm-run-all": "^4.1.5",
-    "stylelint": "^10.1.0",
-    "stylelint-config-vaadin": "^0.1.4",
+    "stylelint": "^14.2.0",
+    "stylelint-config-vaadin": "^0.3.0",
     "wct-browser-legacy": "^1.0.2",
     "wct-istanbul": "^0.14.3",
     "web-component-tester": "^6.9.2"
```

File: src/multiselect-combo-box.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import '@vaadin/input-container/src/vaadin-input-container.js';
+import '@polymer/polymer/lib/elements/dom-repeat.js';
+import '@vaadin/polymer-legacy-adapter/template-renderer.js';
```
Comment (web-padawan) [inline src/multiselect-combo-box.js:2]: I had to add this import to avoid warning about `<template>` usage which is caused by `dom-repeat` 🙈

Comment (gatanaso) [reply src/multiselect-combo-box.js:2]: That's alright 😄

**Comment by srijken** [comment]
Hi, I had to make this change to this PR to avoid an exception. Apparently the __virtualizer can be null:

--- a/src/multiselect-combo-box.js
+++ b/src/multiselect-combo-box.js
@@ -472,7 +472,9 @@ class MultiselectComboBox extends MultiselectComboBoxMixin(InputControlMixin(The
     this.__updateChips();
 
     // Re-render scroller
-    this.$.comboBox.$.dropdown._scroller.__virtualizer.update();
+    if (this.$.comboBox.$.dropdown._scroller.__virtualizer) {
+      this.$.comboBox.$.dropdown._scroller.__virtualizer.update();
+    }
 
     // Wait for chips to render
     requestAnimationFrame(() => {

**Comment by gatanaso** [comment]
@srijken What kind of exception did you have? I recently ran the PR locally and did not get any exceptions?

**Comment by gatanaso** [comment]
@web-padawan This is truly amazing work 💯 , thank you so much for this contribution 🙏

**Comment by srijken** [comment]
@gatanaso about calling `update()` on a null object. Apparently the `__virtualizer` isn't always there? Maybe if the items are loaded later or something?

**Pull request merged**